docs/source/guides/data.rst

```diff
@@ -357,3 +359,21 @@ option when using sequential data.
 
     # Set limit_val_batches as an int
     trainer = Trainer(limit_val_batches=100)
+
+
+*****************************************
+Accessing DataLoaders in LightningModules
+*****************************************
+
+In the case that you require access to the DataLoader or Dataset objects, DataLoaders for each step can be accessed using the `Trainer` object:
+
+.. code-block:: python
+
+    def test_step(batch, batch_idx, dataloader_idx):
+        test_dl = self.trainer.test_dataloaders[dataloader_idx]
+        test_dataset = test_dl.dataset
+        # extract metadata, etc. from the dataset:
+        ...
+
+Since pytorch-lightning automatically takes care of iterating over the DataLoader, this is generally unnecessary unless you need to access metadata
```

Comment:
```suggestion
Since Lightning automatically takes care of iterating over the DataLoader, this is generally unnecessary unless you need to access metadata.
```